### PR TITLE
Use CLE from angr project to load dynamically linked executable

### PR DESCRIPTION
**Pull request by erdnaxe**
This patch proposes to use CLE as a dynamic library loader.

This enables users to load dynamically linked executable. This greatly simplifies the simulation of programs linked with glibc as one would usually only need to hook `calloc`, `malloc` and `free` functions.

To make this patch fully usable, https://github.com/Ledger-Donjon/rainbow/pull/51 is needed.

### CODE REVIEW

**Comment by yhql** [comment]
Thank you very much, very useful.
Do you think `cle` could replace the other wonky loaders (ELF, PE, ... )? Would be a nice cleanup.

File: rainbow/loaders/cleloader.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    # Map memory
+    if verbose:
+        for obj in ld.all_objects:
+            print(f"[ ] Mapping at 0x{obj.min_addr:08X}: {obj.binary_basename}")
+    emu.map_space(ld.min_addr, ld.max_addr, verbose=verbose)
+    for start_addr, backer in ld.memory.backers():
+        emu.emu.mem_write(start_addr, bytes(backer))
+
+    # Load symbols
+    func_symbols = [s for s in ld.symbols if s.is_function and s.is_export]
```
Comment (yhql) [inline rainbow/loaders/cleloader.py:40]: might be useful to default to all functions when available, not just the ones exported

**Comment by erdnaxe** [comment]
> Thank you very much, very useful. Do you think `cle` could replace the other wonky loaders (ELF, PE, ... )? Would be a nice cleanup.

- For ELF loader, CLE seems to handle more cases than current `elfloader`.
- Same for PE loader.
- As far as I know, CLE does not handle a hex loader.

Would it be okay if I propose to drop peloader and elfloader in this pull request?

**Comment by yhql** [comment]
> Would it be okay if I propose to drop peloader and elfloader in this pull request?

Yes you can :)

**Review by yhql — APPROVED** [review]

**Pull request merged**